plugins/sparc2_alignment_data_collector.py

```diff
@@ -0,0 +1,951 @@
+        s_target: Optional[float] = None,
+        z_target: Optional[float] = None
+    ) -> Tuple[float, float, float]:
+        """
+        Execute a single acquisition at the specified mirror (l, s) and stage (z) positions.
+
+        :param filepath: The file path where the acquired data should be saved.
+        :param l_target: Target position for the mirror l-axis.
+        :param s_target: Target position for the mirror s-axis.
+        :param z_target: Target position for the stage z-axis.
+        :return: The current positions acquired (l, s, z).
+        """
+        if l_target is not None and l_target != self.mirror.position.value["l"]:
+            logging.debug(f"Moving to l={l_target:.8f}")
+            self.mirror.moveAbs({"l": l_target}).result()
+        if s_target is not None and s_target != self.mirror.position.value["s"]:
+            logging.debug(f"Moving to s={s_target:.8f}")
+            self.mirror.moveAbs({"s": s_target}).result()
+        if z_target is not None and z_target != self.stage.position.value["z"]:
+            logging.debug(f"Moving to z={z_target:.8f}")
+            self.stage.moveAbs({"z": z_target}).result()
+        # Acquire and process data
+        l_current = self.mirror.position.value["l"]
+        s_current = self.mirror.position.value["s"]
+        x_current = self.mirror_xy.position.value["x"]
+        y_current = self.mirror_xy.position.value["y"]
+        z_current = self.stage.position.value["z"]
+        # TODO: Enable when not using the snapshot controller to save data
+        # data = self.tab._ccd_stream._dataflow.get(asap=False)
+        extra_settings_md = {
+            "l_aligned": self._aligned_pos["l"],
+            "s_aligned": self._aligned_pos["s"],
+            "x_aligned": self._aligned_pos["x"],
+            "y_aligned": self._aligned_pos["y"],
+            "z_aligned": self._aligned_pos["z"],
+            "l": l_current,
+            "s": s_current,
+            "x": x_current,
+            "y": y_current,
+            "z": z_current,
+            "dl": self._aligned_pos["l"] - l_current,
+            "ds": self._aligned_pos["s"] - s_current,
+            "dx": self._aligned_pos["x"] - x_current,
+            "dy": self._aligned_pos["y"] - y_current,
+            "dz": self._aligned_pos["z"] - z_current,
+            "wd": self.ebeam_focus.position.value["z"],
+        }
+        if l_target is not None:
+            extra_settings_md.update({"l_target": l_target, "dl_target": l_target - l_current})
+        if s_target is not None:
+            extra_settings_md.update({"s_target": s_target, "ds_target": s_target - s_current})
+        if z_target is not None:
+            extra_settings_md.update({"z_target": z_target, "dz_target": z_target - z_current})
+        self.snapshot_controller.snapshot_viewport(
+            self.tab, filepath, self._exporter, anim=False, forcemd={model.MD_EXTRA_SETTINGS: extra_settings_md}
+        )
```

Comment:
Please avoid using the high level `snapshot_viewport` to save an image. The code below should work as well. Assuming the stream is playing, you don't have to call `_onNewData()`, and `raw_data = self.tab._ccd_stream.raw` should return the lastest acquire data. (then, you can update its metadata, as you currently do in `snapshot_viewport`)